### PR TITLE
Don't break attributes over multiple lines.

### PR DESCRIPTION
**Pull request by IwanKaramazow**
Improves attribute formatting:
* doesn't break attributes on multiple lines, when it isn't necessary.
* provides better identation when breaking the declaration over multiple lines

Example rehydrate, before:
```rust
external createElement : reactClass => props::Js.t {..}? => array reactElement => reactElement = "createElement" [@@bs.splice] [@@bs.val] [@@bs.module
                                                                    "react"
                                                                    ];

external createClassInternalHack : Js.t 'classSpec => reactClass = "createClass" [@@bs.val] [@@bs.module
                                                                    "react"
                                                                    ];

external createCompositeElementInternalHack : reactClass =>
                                              Js.t {.. reasonProps : 'props} =>
                                              array reactElement =>
                                              reactElement = "createElement" [@@bs.val] [@@bs.module
                                                                    "react"
                                                                    ] [@@bs.splice];
```

After:
```rust
external createElement : reactClass => props::Js.t {..}? => array reactElement => reactElement =
  "createElement" [@@bs.splice] [@@bs.val] [@@bs.module "react"];

external createClassInternalHack : Js.t 'classSpec => reactClass =
  "createClass" [@@bs.val] [@@bs.module "react"];

type reactElementthisIsasupersupersupersupserupserupserupseruplongtype;

external createCompositeElementInternalHack :
  reactClass =>
  Js.t {.. reasonProps : 'props} =>
  array reactElementthisIsasupersupersupersupserupserupserupseruplongtype =>
  reactElement =
  "createElement" [@@bs.val] [@@bs.module "react"] [@@bs.splice];

external createCompositeElementInternalHack :
  reactClass => Js.t {.. reasonProps : 'props} => array reactElement => reactElement =
  "createElement" [@@bs.val] [@@bs.module "react"] [@@bs.splice];
```

### CODE REVIEW

**Comment by yunxing** [comment]
Looks good. Let's wait for the CI.

**Comment by chenglou** [comment]
Should these attributes break on many lines? They can contain arbitrary payloads. Seems like they should be arranged like function arguments currently?

**Comment by IwanKaramazow** [comment]
I think this is ok for now, `#structure_item` will handle the output with breaking from more complex structures.

**Comment by yunxing** [comment]
@chenglou I think putting them into a single line makes some sense, as in we always want to view the attributes with the node it's decorating? 

Altnative is to format the example as 

```
external createClassInternalHack : Js.t 'classSpec => reactClass = "createClass" [@@bs.val]          
                                                                            [@@bs.module "react"];
```
which I think is also acceptable.

**Comment by IwanKaramazow** [comment]
How should I proceed?

**Comment by chenglou** [comment]
What if they're long? 3 Tags for BS isn't _too_ uncommon. They'd break the column limit hard. Why not

```
external createClassInternalHack : Js.t 'classSpec => reactClass = "createClass" 
  [@@bs.val] [@@bs.module "react"];
```

**Comment by IwanKaramazow** [comment]
I kinda like the one liner more, but agreed it does make sense to break the attributes.

@SanderSpies, @jordwalke  What do you think?

**Comment by chenglou** [comment]
No hard opinion on this fwiw. Though maybe this'd work better than the previous example:

```
external createClassInternalHack : Js.t 'classSpec => reactClass = 
 "createClass" [@@bs.val] [@@bs.module "react"];
```

**Comment by yunxing** [comment]
@chenglou I like your example better.

@IwanKaramazow Let us know if it is hard to implement this (you would need to make a Label before and after the "=", which could be not that trivial). If it is hard, let's just push what we have currently.

**Comment by IwanKaramazow** [comment]
@yunxing I think I made @chenglou 's version work.
See latest commit.

Example of refmt on rehydrate:
```ocaml
external createElement : reactClass => props::Js.t {..}? => array reactElement => reactElement =
  "createElement" [@@bs.splice] [@@bs.val] [@@bs.module "react"];

external createClassInternalHack : Js.t 'classSpec => reactClass =
  "createClass" [@@bs.val] [@@bs.module "react"];

type reactElementthisIsasupersupersupersupserupserupserupseruplongtype;

external createCompositeElementInternalHack :
  reactClass =>
  Js.t {.. reasonProps : 'props} =>
  array reactElementthisIsasupersupersupersupserupserupserupseruplongtype =>
  reactElement =
  "createElement" [@@bs.val] [@@bs.module "react"] [@@bs.splice];

external createCompositeElementInternalHack :
  reactClass => Js.t {.. reasonProps : 'props} => array reactElement => reactElement =
  "createElement" [@@bs.val] [@@bs.module "react"] [@@bs.splice];
```

**Comment by yunxing** [comment]
@IwanKaramazow Thanks, looking into this now.

**Comment by IwanKaramazow** [comment]
@yunxing any remarks?

File: src/reason_pprint_ast.ml [hunk 1]
```diff
@@ -5065,6 +5065,36 @@ class printer  ()= object(self:'self)
...
+      label ~space:true
+        (makeList ~postSpace:true
+          [atom "external"; protectIdentifier vd.pval_name.txt; atom ":"])
+        (self#core_type vd.pval_type)
+    in
+    let frstHalf = makeList ~postSpace:true [lblBefore; atom "="] in
+    let string_literals = makeSpacedBreakableInlineList (List.map self#constant_string vd.pval_prim) in
+    (* when there aren't any attrs, add a final semi here *)
+    let prim = if (List.length attrs) == 0
+      then appendSep false ";" string_literals
```
Comment (yunxing) [inline src/reason_pprint_ast.ml]: Took me a while to understand why you add a ";" here.  I think it is probably due to some tweaks we put into comment interleaving logic?

**Comment by yunxing** [comment]
@IwanKaramazow Sorry for the delay. Diff looks good. I think I should review the usage of appendSep logic a bit.

**Pull request merged**